universal-application-tool-0.0.1/app/views/dev/FileUploadView.java

```diff
@@ -22,43 +19,58 @@
 import models.StoredFile;
 import play.mvc.Http.Request;
 import play.twirl.api.Content;
-import services.aws.SignedS3UploadRequest;
-import services.aws.SimpleStorage;
+import services.cloud.StorageClient;
+import services.cloud.StorageUploadRequest;
 import views.BaseHtmlLayout;
 import views.BaseHtmlView;
 import views.HtmlBundle;
+import views.ViewUtils;
 import views.style.Styles;
 
-/** Renders a page for a developer to test uploading files. */
+/**
+ * Renders a page for a developer to test uploading files. The logic for creating the file upload
+ * form has been extracted out to {@link FileUploadViewStorageStrategy}.
+ */
 public class FileUploadView extends BaseHtmlView {
+
   private final BaseHtmlLayout layout;
-  private final SimpleStorage s3Client;
+  private final StorageClient storageClient;
+  private final ViewUtils viewUtils;
+  private final FileUploadViewStorageStrategy storageStrategy;
 
   @Inject
-  public FileUploadView(BaseHtmlLayout layout, SimpleStorage s3Client) {
+  public FileUploadView(
+      BaseHtmlLayout layout,
+      StorageClient storageClient,
+      ViewUtils viewUtils,
+      FileUploadViewStorageStrategy storageStrategy) {
     this.layout = checkNotNull(layout);
-    this.s3Client = checkNotNull(s3Client);
+    this.storageClient = checkNotNull(storageClient);
+    this.viewUtils = checkNotNull(viewUtils);
+    this.storageStrategy = checkNotNull(storageStrategy);
   }
 
   public Content render(
       Request request,
-      SignedS3UploadRequest signedRequest,
+      StorageUploadRequest signedRequest,
       ImmutableList<StoredFile> files,
       Optional<String> maybeFlash) {
     String title = "Dev file upload";
-    HtmlBundle bundle =
-        layout
-            .getBundle()
-            .setTitle(title)
-            .addMainContent(
-                div()
-                    .with(div(maybeFlash.orElse("")))
-                    .with(h1(title))
-                    .with(div().with(fileUploadForm(signedRequest)))
-                    .with(
-                        div()
-                            .withClasses(Styles.GRID, Styles.GRID_COLS_2)
-                            .with(div().with(h2("Current Files:")).with(pre(renderFiles(files))))));
+    ContainerTag fileUploadForm;
+
+    HtmlBundle bundle = layout.getBundle();
+    fileUploadForm = storageStrategy.getFileUploadForm(viewUtils, signedRequest, bundle);
```

Comment:
suggestion (blocking): seeing that this is where the `null` would end up from the strategy classes, I suggest throwing an exception instead of returning `null` or `Optional`

Reply:
I've updated so that the strategy patterns throw RuntimeExceptions, can you take a look and let me know if the implementation looks okay?